app/models/assignment_repo/creator.rb

```diff
@@ -189,5 +190,63 @@ def suffixed_repo_name(repository_name, suffix_count)
       suffix = "-#{suffix_count}"
       repository_name.truncate(100 - suffix.length, omission: '') + suffix
     end
+
+    # Handles assignment repository setup
+    #
+    # configs_repo    - GitHubRepository containing the configuration files
+    # assignment_repo - GitHubRepository for which to perform the configuration
+    #                   setups
+    #
+    # Returns nothing
+    def setup_repository(configs_repo, assignment_repo)
+      return unless configs_repo.classroom_configs?
+      conf_tree = configs_repo.branch('github-classroom').commit.commit.tree.sha
+      configs_repo.tree(conf_tree).tree.each do |conf|
+        process_config(configs_repo, assignment_repo, conf)
+      end
+
+      wait_import_completion assignment_repo
+
+      assignment_repo.remove_branch('github-classroom')
+    end
+
+    def wait_import_completion(github_repository)
```

Comment:
[Article on how Dynos work](https://devcenter.heroku.com/articles/dynos) for reference.

@nwoodthorpe we have two web dynos running atm running puma. Each of those puma servers are running 4 different threads [`RAILS_MAX_THREADS`](https://github.com/education/classroom/blob/master/config/puma.rb#L9)

So we _should_ be ok hanging for a bit, I don't know about 2 minutes though.